File: CSharp/SpecStudioExamplesTestsCSharp/common/PostTyped.cs
namespace SpecStudioExamplesTestsCSharp.common
{
using System;
using System.Buffers;
using System.Collections.Generic;
using System.Globalization;
using System.Text;
using System.Text.Json;
using production;

    public class PostTyped
    {
        public int userId;
        public int id;
        public string title;
        public string body;

        public PostTyped(int userId, int id, string title, string body)
        {
            this.userId = userId;
            this.id = id;
            this.title = title;
            this.body = body;
        }

        public PostString ToPostString()
        {
            return new PostString(
                Json.ToText(this.userId),
                Json.ToText(this.id),
                Json.ToText(this.title),
                Json.ToText(this.body)
            );
        }

        public static List<PostString> ToStringList(List<PostTyped> list)
        {
            var result = new List<PostString>();
            foreach (var t in list) result.Add(t.ToPostString());
            return result;
        }

        public static List<PostTyped> FromStringList(List<PostString> list)
        {
            var result = new List<PostTyped>();
            foreach (var s in list) result.Add(s.ToPostTyped());
            return result;
        }

        public void WriteJson(Utf8JsonWriter w)
        {
            w.WriteStartObject();
            w.WriteNumber("userId", this.userId);
            w.WriteNumber("id", this.id);
            w.WriteString("title", this.title);
            w.WriteString("body", this.body);
            w.WriteEndObject();
        }

        public string ToJSON()
        {
            var buffer = new ArrayBufferWriter<byte>();
            using (var w = new Utf8JsonWriter(buffer)) { WriteJson(w); }
            return Encoding.UTF8.GetString(buffer.WrittenSpan);
        }

        public static PostTyped FromJsonElement(JsonElement m)
        {
            return new PostTyped(
                Json.AsInt(Json.Require(m, "userId"), "userId"),
                Json.AsInt(Json.Require(m, "id"), "id"),
                Json.AsString(Json.Require(m, "title"), "title"),
                Json.AsString(Json.Require(m, "body"), "body")
            );
        }

        public static PostTyped FromJSON(string json)
        {
            return FromJsonElement(Json.Parse(json));
        }

        public static string ToJSONList(List<PostTyped> list)
        {
            var buffer = new ArrayBufferWriter<byte>();
            using (var w = new Utf8JsonWriter(buffer))
            {
                w.WriteStartArray();
                foreach (var item in list) item.WriteJson(w);
                w.WriteEndArray();
            }
            return Encoding.UTF8.GetString(buffer.WrittenSpan);
        }

        public static List<PostTyped> FromJSONList(string json)
        {
            var result = new List<PostTyped>();
            var root = Json.Parse(json);
            Json.RequireArray(root, "PostTyped");
            foreach (var e in root.EnumerateArray()) result.Add(FromJsonElement(e));
            return result;
        }

        public override string ToString()
        {
            return $"userId={userId}, id={id}, title={title}, body={body}";
        }

        public override bool Equals(object? obj)
        {
            if (obj is not PostTyped other) return false;
            return object.Equals(this.userId, other.userId)
                && object.Equals(this.id, other.id)
                && object.Equals(this.title, other.title)
                && object.Equals(this.body, other.body);
        }

        public override int GetHashCode()
        {
            var h = new System.HashCode();
            h.Add(this.userId);
            h.Add(this.id);
            h.Add(this.title);
            h.Add(this.body);
            return h.ToHashCode();
        }
    }
}

File: Java/SpectableJavaExamplesTest/src/test/java/spectable/common/PostTyped.java
package spectable.common;

import java.util.ArrayList;
import java.util.List;
import java.util.LinkedHashMap;
import java.util.Map;
import java.util.Objects;
import production.*;
import records.*;
import calculator.*;

public class PostTyped {
    public int userId;
    public int id;
    public String title;
    public String body;

    public PostTyped(int userId, int id, String title, String body) {
        this.userId = userId;
        this.id = id;
        this.title = title;
        this.body = body;
    }

    public PostTyped(PostString s) {
        this.userId = Integer.parseInt(s.userId);
        this.id = Integer.parseInt(s.id);
        this.title = s.title;
        this.body = s.body;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof PostTyped)) return false;
        PostTyped that = (PostTyped) o;
        return Objects.equals(userId, that.userId)
            && Objects.equals(id, that.id)
            && Objects.equals(title, that.title)
            && Objects.equals(body, that.body);
    }

    @Override
    public int hashCode() {
        return Objects.hash(userId, id, title, body);
    }

    public PostString toPostString() {
        return new PostString(String.valueOf(userId), String.valueOf(id), String.valueOf(title), String.valueOf(body));
    }

    public static List<PostTyped> fromStringList(List<PostString> list) {
        List<PostTyped> result = new ArrayList<>();
        for (PostString s : list) result.add(new PostTyped(s));
        return result;
    }

    public static List<PostString> toStringList(List<PostTyped> list) {
        List<PostString> result = new ArrayList<>();
        for (PostTyped t : list) result.add(t.toPostString());
        return result;
    }

    /** Internal plumbing for {@link Json}; use toJSON() for JSON text. */
    public Map<String, Object> toJsonValue() {
        Map<String, Object> m = new LinkedHashMap<String, Object>();
        m.put("userId", userId);
        m.put("id", id);
        m.put("title", title);
        m.put("body", body);
        return m;
    }

    public String toJSON() { return Json.write(toJsonValue()); }

    /** Internal plumbing for {@link Json}; use fromJSON(String) for JSON text. */
    public static PostTyped fromJsonValue(Map<String, Object> m) {
        if (m == null) return null;
        return new PostTyped(
            Json.asInt(Json.require(m, "userId"), "userId"),
            Json.asInt(Json.require(m, "id"), "id"),
            Json.asString(Json.require(m, "title"), "title"),
            Json.asString(Json.require(m, "body"), "body"));
    }

    public static PostTyped fromJSON(String json) {
        return fromJsonValue(Json.parseObject(json));
    }

    public static String toJSONList(List<PostTyped> list) {
        List<Object> arr = new ArrayList<Object>();
        if (list != null)
            for (PostTyped item : list) arr.add(item == null ? null : item.toJsonValue());
        return Json.write(arr);
    }

    public static List<PostTyped> fromJSONList(String json) {
        List<PostTyped> result = new ArrayList<PostTyped>();
        for (Object e : Json.parseArray(json))
            result.add(fromJsonValue(Json.asObject(e, "PostTyped")));
        return result;
    }
}
